src/internal/connector/onedrive/collections_test.go

```diff
@@ -2665,3 +2666,83 @@ func (suite *OneDriveCollectionsUnitSuite) TestCollectItems() {
 		})
 	}
 }
+
+func (suite *OneDriveCollectionsUnitSuite) TestURLCacheIntegration() {
+	driveID := "test-drive"
+	collCount := 3
+
+	table := []struct {
+		name             string
+		items            []deltaPagerResult
+		deltaURL         string
+		prevDeltaSuccess bool
+		prevDelta        string
+		err              error
+	}{
+		{
+			name: "cache is attached",
+		},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			mbh := mock.DefaultOneDriveBH()
+			mbh.ItemPagerV = map[string]api.DriveItemEnumerator{}
+
+			c := NewCollections(
+				mbh,
+				"test-tenant",
+				"test-user",
+				testFolderMatcher{(&selectors.OneDriveBackup{}).Folders(selectors.Any())[0]},
+				nil,
+				control.Options{ToggleFeatures: control.Toggles{}})
+
+			if _, ok := c.CollectionMap[driveID]; !ok {
+				c.CollectionMap[driveID] = map[string]*Collection{}
+			}
+
+			// Add a few collections
+			for i := 0; i < collCount; i++ {
+				coll, err := NewCollection(
+					&itemBackupHandler{api.Drives{}},
+					nil,
+					nil,
+					driveID,
+					nil,
+					control.Options{ToggleFeatures: control.Toggles{}},
+					CollectionScopeFolder,
+					true,
+					nil)
+				require.NoError(t, err, clues.ToCore(err))
+
+				c.CollectionMap[driveID][strconv.Itoa(i)] = coll
+				require.Equal(t, nil, coll.cache, "cache not nil")
+			}
+
+			err := c.addURLCacheToDriveCollections(
+				ctx,
+				driveID,
+				fault.New(true))
+
+			require.NoError(t, err, clues.ToCore(err))
+
+			// Check that all collections have the same cache instance attached
+			// to them
+			var uc *urlCache
```

Comment:
Does this need to be the underlying type, or could it be the interface?  The latter seems more resilient across change than type-casting the implementing struct from the coll.cache.

Reply:
I used the underlying type to confirm that the same URL cache instance is shared by all collections.